dask/dataframe/core.py

```diff
@@ -11,6 +11,7 @@
 
 import numpy as np
 import pandas as pd
+from pandas.api.extensions import no_default as pd_no_default
```

Comment:
As is the changes here will pass the `no_default` from `pandas` down to the `.dropna(...)` method applied to each DataFrame partition. For Dask DataFrames that are backed by `pandas` DataFrames this should work totally fine, but my guess is this will break for `cudf`. cc @galipremsagar @quasiben for thoughts. 

xref https://github.com/dask/dask/issues/8645